pkg/kp/flags/kingpin.go

```diff
@@ -29,10 +33,31 @@ func ParseWithConsulOptions() (string, kp.Options) {
 		}
 		*token = string(tokenBytes)
 	}
+	var transport http.RoundTripper
+	if *caFile != "" || *keyFile != "" || *certFile != "" {
```

Comment:
I'm not sure why you want all or nothing. It seems reasonable to me that one might use the default certificate authorities while supplying a client certificate.